include/art/art.hpp

```diff
@@ -27,7 +27,7 @@ template <class T> class art {
    * @param key - The key to find.
    * @return the value associated with the key or a nullptr.
    */
-  T *get(const char *key) const;
+  T get(const char *key) const;
```

Comment:
I have suggested a documentation change to make this change explicit, but I am also curious to know your motivations in returning value rather than pointer. 
On one hand it's good since user will have to store explicitly pointers, on the other hand it requires the stored object to have default constructor.
I am happy with this change, but since I am still learning c++, I would like to know the reasons hoping i will lean more best practices from you.

Reply:
The motivation is to give the developer an additional memory management option. Until now the underlying was a pointer. Some prefer [smart pointers](https://en.cppreference.com/book/intro/smart_pointers) for their use case, this api change brings support for smart pointers. Those who want traditional pointers still can have them by providing a pointer type as a template argument (e.g., `art<int*>`). Those who want smart pointers use a smart pointer template argument instead (e.g., `art<shared_ptr<int>>`).